src/snowflake/cli/plugins/connection/util.py

```diff
@@ -50,11 +56,46 @@ def is_regionless_redirect(conn: SnowflakeConnection) -> bool:
         *_, cursor = conn.execute_string(REGIONLESS_QUERY, cursor_class=DictCursor)
         return cursor.fetchone()["REGIONLESS"].lower() == "true"
     except:
-        # by default, assume that
         log.warning("Cannot determine regionless redirect; assuming True.")
         return True
 
 
+def guess_regioned_host_from_allowlist(conn: SnowflakeConnection) -> str | None:
+    """
+    Use SYSTEM$ALLOWLIST to find a regioned host (<account>.x.y.z.snowflakecomputing.com)
+    that corresponds to the given Snowflake connection object.
+    """
+    *_, cursor = conn.execute_string(ALLOWLIST_QUERY, cursor_class=DictCursor)
+    allowlist_tuples = json.loads(cursor.fetchone()["SYSTEM$ALLOWLIST()"])
+    for t in allowlist_tuples:
+        if t["type"] == SNOWFLAKE_DEPLOYMENT:
+            host_parts = t["host"].split(".")
+            if len(host_parts) == 6:
+                return t["host"]
+    return None
+
+
+def get_region(conn: SnowflakeConnection) -> str:
+    """
+    Given a connection, determine the region of the account in the
+    form "x.y.z", or raise an error if we could not find an appropriate host.
+    """
+    if conn.host:
+        host_parts = conn.host.split(".")
+        if host_parts[-1] == "local":
```

Comment:
can we add a test to this since it is missing coverage?

Reply:
Done!